doc/api/process.markdown

```diff
@@ -59,6 +59,9 @@ finished running the process will exit. Therefore you **must** only perform
 checks on the module's state (like for unit tests). The callback takes one
 argument, the code the process is exiting with.
 
+This event is not guaranteed to fire in all situations. For example, if the
+process receives `SIGINT`, it may exit without firing this event.
```

Comment:
good point. that's still just another user shutdown like `ctrl` + `c` stuff. here's a list:

http://www.gnu.org/software/libc/manual/html_node/Termination-Signals.html